tomviz/operators/OperatorPython.cxx

```diff
@@ -484,13 +484,68 @@ QJsonObject OperatorPython::serialize() const
   return json;
 }
 
+namespace {
+QJsonObject findJsonObject(const QJsonArray& array, const QString& name)
+{
+  for (int i = 0; i < array.size(); ++i) {
+    auto object = array[i].toObject();
+    if (object["name"].toString() == name) {
+      return object;
+    }
+  }
+  return QJsonObject();
+}
+
+QVariant castJsonArg(const QJsonValue& arg, const QString& type)
+{
+  if (arg.isArray()) {
+    auto arr = arg.toArray();
+    QVariantList arrayList;
+    if (type == "int" || type == "enumeration") {
+      for (int i = 0; i < arr.size(); ++i) {
+        arrayList << arr[i].toInt();
+      }
+    } else if (type == "double") {
+      for (int i = 0; i < arr.size(); ++i) {
+        arrayList << arr[i].toDouble();
+      }
```

Comment:
I think we can avoid duplicating the int  || enumeration and double blocks by recursively calling ```castJsonArg ``` for each element in the array?

Reply:
Agreed, especially if we want to get to nested arrays etc. Merging what we have for now.